Pre-commit failing for all branches.

I noticed that the pre-commit fails for all branches.
In the run pre-commit stage, I get the following error:
```
[INFO] Initializing environment for https://github.com/ambv/black.
[INFO] Initializing environment for https://github.com/pre-commit/pre-commit-hooks.
[INFO] Initializing environment for https://github.com/pycqa/flake8.
[INFO] Initializing environment for local.
[INFO] Initializing environment for https://github.com/pre-commit/mirrors-mypy.
[INFO] Initializing environment for https://github.com/pre-commit/mirrors-mypy:pydantic~=1.10,types-all.
[INFO] Initializing environment for https://github.com/PyCQA/isort.
[INFO] Installing environment for https://github.com/ambv/black.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
An unexpected error has occurred: CalledProcessError: command: ('/opt/hostedtoolcache/Python/3.10.8/x64/bin/python', '-mvirtualenv', '/home/runner/.cache/pre-commit/repo58o8v0_a/py_env-python3.8', '-p', 'python3.8')
return code: 1
expected return code: 0
stdout:
    RuntimeError: failed to find interpreter for Builtin discover of python_spec='python3.8'
    
stderr: (none)
Check the log at /home/runner/.cache/pre-commit/pre-commit.log
Error: Process completed with exit code 1.
```

If I understand it correctly, the github servers no longer have python 3.8 installed by default, so we will have to either specify that it should be installed in our virtual environment or update the references in the script to python 3.10.

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -49,10 +49,10 @@ jobs:
       with:
         submodules: true
 
-    - name: Set up Python 3.10
+    - name: Set up Python 3.8
       uses: actions/setup-python@v4
       with:
-        python-version: '3.10'
+        python-version: '3.8'
 
     - name: Install dependencies
       run: |

